docs/contributor/draft/rate-limit-api.md

```diff
@@ -0,0 +1,121 @@
+# Rate Limit Proposal
+
+## Spec 
+
+| field                                            | type        | description                                                     | required                      |
+|--------------------------------------------------|-------------|-----------------------------------------------------------------|-------------------------------|
+| wokloadSelector                                  | object      | in case of empty, it's applied for the whole namespace          | no                            |
+| workloadSelector.labels                          | object      | allows to select given workload by a selector                   | no                            |
+| localRateLimit                                   | object      | allows to describe local rate limit properties                  | yes                           |
+| localRateLimit.by                                | object      | container for criteria you can limit by (for the future)        | yes                           |
+| localRateLimit.by.requests                       | object      | allows to set rate limit based on plain number of requests      | yes                           |
+| localRateLimit.by.requests.time                  | object      | allows to describe local rate limiting using time periods       | yes (if buckets not defined)  |
```

Comment:
Let's start with exposing only time then, it won't be eventually a lot of work in the future to expose bucket approach in case of any feature request.